Services/Component/classes/class.ilPluginRawReader.php

```
@@ -0,0 +1,24 @@
+<?php
+/* Copyright (c) 2020 Daniel Weise <daniel.weise@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+declare(strict_types=1);
+
+class ilPluginRawReader
+{
+    const BASE_PLUGIN_PATH = 'Customizing/global/plugins';
+    const SEARCH_PATTERN = 'plugin.php';
+
+    public function getPluginNames() : ?\Iterator
+    {
+        if (!@is_dir(self::BASE_PLUGIN_PATH)) {
+            return null;
```

Maybe we would want to throw an exception here? Why should the `BASE_PLUGIN_PATH` be inexistent and ain't it something we should notify our user about?

---

done